### PR TITLE
BUG: fix return type of covers/covered_by

### LINKED ISSUE #1829: Failures for GeoSeries.covers (object dtype instead of bool) in the dev build
We have a few failures in the dev build, the three `covers` related tests are failing:

```
 =================================== FAILURES ===================================
______________________ TestGeomMethods.test_covers_itself ______________________
[gw0] linux -- Python 3.8.6 /usr/share/miniconda/envs/test/bin/python

self = <geopandas.tests.test_geom_methods.TestGeomMethods object at 0x7f39052dcfa0>

    def test_covers_itself(self):
        # Each polygon in a Series covers itself
        res = self.g1.covers(self.g1)
        exp = Series([True, True])
>       assert_series_equal(res, exp)
E       AssertionError: Attributes of Series are different
E       
E       Attribute "dtype" are different
E       [left]:  object
E       [right]: bool

geopandas/tests/test_geom_methods.py:426: AssertionError
_________________________ TestGeomMethods.test_covers __________________________
[gw0] linux -- Python 3.8.6 /usr/share/miniconda/envs/test/bin/python

self = <geopandas.tests.test_geom_methods.TestGeomMethods object at 0x7f39050a8250>

    def test_covers(self):
        res = self.g7.covers(self.g8)
        exp = Series([True, False])
>       assert_series_equal(res, exp)
E       AssertionError: Attributes of Series are different
E       
E       Attribute "dtype" are different
E       [left]:  object
E       [right]: bool

geopandas/tests/test_geom_methods.py:431: AssertionError
_____________________ TestGeomMethods.test_covers_inverse ______________________
[gw0] linux -- Python 3.8.6 /usr/share/miniconda/envs/test/bin/python

self = <geopandas.tests.test_geom_methods.TestGeomMethods object at 0x7f3905175220>

    def test_covers_inverse(self):
        res = self.g8.covers(self.g7)
        exp = Series([False, False])
>       assert_series_equal(res, exp)
E       AssertionError: Attributes of Series are different
E       
E       Attribute "dtype" are different
E       [left]:  object
E       [right]: bool

geopandas/tests/test_geom_methods.py:436: AssertionError
```

I haven't been able to reproduce the issue locally (I tested both with latest pandas master, as with numpy nightly wheel). For some reason, it started to return an object dtype Series, instead of boolean Series (but it's also strange that only `covers` has this issue, and not any of the other predicates)

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Maybe closes #1829 (can't reproduce it locally, so also can't test if this helps. But this seems wrong in any case)

### CODE REVIEW

File: geopandas/base.py [hunk 1]
```diff
@@ -894,7 +894,7 @@ def covers(self, other):
         other : Geoseries or geometric object
             The Geoseries (elementwise) or geometric object to check is being covered.
         """
-        return _binary_geo("covers", self, other)
+        return _binary_op("covers", self, other)
```
Comment (jorisvandenbossche) [inline geopandas/base.py:897]: All the other predicates use `_binary_op`, as `_binary_geo` feeds the result to a `GeoSeries(...)` constructor

**Comment by martinfleis** [comment]
I made this fix already in #1668. Trying to figure out now if it is fixed there or not but had to update the PR first.

**Comment by martinfleis** [comment]
> I made this fix already in #1668. Trying to figure out now if it is fixed there or not but had to update the PR first.

Yep, fixed there same as here.

**Comment by jorisvandenbossche** [comment]
Ah, and we actually also already had an issue about this: https://github.com/geopandas/geopandas/issues/1701

**Pull request closed without merging**